### PR TITLE
RDE 3 - infra: global command flag handling

### PR DESCRIPTION
**Pull request by molnarm**
Global flag handling improvements: `output`, `theme`, `quiet` etc. registered

_This is PR 3 of 15, splitting the changes into reviewable chunks. PRs 1-4 are just infrastructure._

### CODE REVIEW

File: cli/cli_test.go [hunk 1]
```diff
@@ -257,3 +394,78 @@ func Test_before_calledWithoutExecute_doesNotPanic(t *testing.T) {
...
+		})
+	}
+}
+
+// Test_ymlMergeOutputFlag_ShadowsGlobalOutputFlag guards the shorthand
+// collision called out in the master rde-migration plan: yml merge's own
+// --output/-o (an output directory) must keep working unchanged once the
+// root gains a persistent --output/-o (an output format) — cobra's flag
+// merge order (local flags are registered before parent persistent flags are
+// merged in) makes the local one win, with no panic on the shared shorthand.
```
Comment (godrei) [inline cli/cli_test.go]: > with no panic

I think this is only true if the flags (long) name matches, otherwise (like `-o` shorthand for `--outputfile`) it panics

```
// AddFlagSet adds one FlagSet to another. If a flag is already present in f
// the flag from newSet will be ignored.
func (f *FlagSet) AddFlagSet(newSet *FlagSet) {
	if newSet == nil {
		return
	}
	newSet.VisitAll(func(flag *Flag) {
		if f.Lookup(flag.Name) == nil {
			f.AddFlag(flag)
		}
	})
}
```

Comment (molnarm) [reply cli/cli_test.go]: Good catch!
Updated the comment (explanation was incorrect): this works today because -o stands for --output everywhere. Also added a test to check every command for collisions with global flags, there are none currently.

**Review by godrei — APPROVED** [review]
